livekit-ffi/src/conversion/room.rs

```diff
@@ -213,8 +213,9 @@ impl From<proto::AudioEncoding> for AudioEncoding {
 }
 
 impl From<&FfiRoom> for proto::RoomInfo {
-    fn from(value: &FfiRoom) -> Self {
+    #[tokio::main]
+    async fn from(value: &FfiRoom) -> Self {
         let room = &value.inner.room;
-        Self { sid: room.sid().into(), name: room.name(), metadata: room.metadata() }
+        Self { sid: room.sid().await.into(), name: room.name(), metadata: room.metadata() }
```

Comment:
I think we should create a new FfiRequest to get the SID as it is now async

Example of an async request: https://github.com/livekit/rust-sdks/blob/8b276f9d4b98437a139e1cbe41cfda1332ce1120/livekit-ffi/protocol/room.proto#L120

Reply:
how would that help this line? `room.sid()` is still asynchronous, so `impl From` has to be async as well

Reply:
So I think it could also be set to None and we can notify the FfiClient once we have a correct SID?

Reply:
could we make `inner.sid` an `Option<RoomSid>`, then? `(*Room).sid()` would still return an unwrapped sid when available but i think it makes things easier if we used an option here

Reply:
Ideally the field will just look like sid: Promise<RoomSid>.
I think it's correct, you may need Option<T> internally

Reply:
So I was thinking that we could add a new Promise utility to the livekit-protocol crate:
Something similar to this Go implementation: https://github.com/livekit/protocol/blob/main/utils/promise.go
So we don't have to put sid_rx and sid_tx inside the room implementation, and we can keep things cleaner there.

Reply:
added `Promise<T>` to protocol.
it doesn't work *exactly* like JS Promises since that's not what we need here, it's mostly just an async wrapper for `tokio::sync::oneshot`, but it is a lot cleaner.

Reply:
Nice, I added a few more comments